pkg/driver/node_server.go

```diff
@@ -1621,6 +1623,28 @@ func (driver *Driver) NodeGetInfo(ctx context.Context, req *csi.NodeGetInfoReque
 		nodeMaxVolumesLimit = defaultMaxVolPerNode
 	}
 
+	// Enable watcher only once. GetNodeInfo rpc may be called multiple times by external
+	// provisioner.
+	if !isWatcherEnable {
+		// Create a anonymous wrapper function over nodeGetInfo. This is os event driven
+		// fn execution.
+		getNodeInfoFunc := func() {
+			nodeID, err := driver.nodeGetInfo()
+			if err != nil {
+				log.Errorf("Failed to update %s nodeInfo. Error: %s", nodeID, err.Error())
+			}
+			return
+		}
```

Comment:
I feel this is overkill.  The watcher can determine which files have changed and therefore limit how much of the node info needs to be updated.  For example, why update the entire node when only the iqn has changed.  Likewise, why update the entire node when only the network has changed?

This might be difficult to achieve so at least minimizing the frequency with which this function is called would probably be enough.  So can you add a logger to this function so we can start gauging that?

Reply:
We aren't updating the entire HPENodeInfo if you check LoadNodeInfo() in flavor.go, only specific changes are updated. So if there is only network change we do network update only. Please check and let me know.

Reply:
Right.  That's LoadNodeInfo.  But that is preceded by nodeGetInfo which calls driver.chapiDriver.GetHostNameAndDomain, driver.chapiDriver.GetHostInitiators, and driver.chapiDriver.GetHostNetworks.  It seems we should only invoke CHAPI when necessary.

Reply:
Yes, I think we should also consider that this is event-driven and we aren't expecting a regular update to Network, Initiators or Domain, etc. So, even if we call nodeGetInfo() that should be okay since it will not have a major performance impact. Thought?

Reply:
Ya, that's fair.  I don't think this is a hard requirement, just ideal.  I think if we limit the overall frequency at which we update (only when necessary), this should be fine.